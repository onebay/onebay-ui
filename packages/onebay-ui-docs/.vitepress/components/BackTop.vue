<template>
  <div :class="wrapperCls" @click="backTop">
    <slot>
      <svg
        t="1606313325998"
        class="icon"
        viewBox="0 0 1024 1024"
        version="1.1"
        xmlns="http://www.w3.org/2000/svg"
        p-id="1414"
        width="64"
        height="64"
      >
        <path
          d="M522.439258 279.608186 69.730643 732.316801 975.145827 732.316801Z"
          p-id="1415"
          fill="#3eaf7c"
        ></path>
      </svg>
    </slot>
  </div>
</template>
<script>
import { defineComponent, computed } from "vue";
import classNames from "classnames";
import useScrollTop from "../hooks/useScrollTop";

const prefixCls = "ob-back-top";

export default defineComponent({
  name: "BackTop",
  props: {
    className: {
      type: String,
      default: "",
    },
    distance: {
      type: Number,
      default: 100,
    },
  },
  emits: ["click"],
  setup(props, { slots, emit }) {
    const scrollTop = useScrollTop();
    const show = computed(() => scrollTop.value > props.distance);

    const backTop = (e) => {
      window.scrollTo({ top: 0, behavior: "smooth" });
      emit("click", e);
    };

    const wrapperCls = computed(() => {
      return classNames(prefixCls, props.className, {
        [`${prefixCls}__hide`]: !show.value,
      });
    });

    return {
      backTop,
      wrapperCls,
    };
  },
});
</script>

<style lang="scss" scoped>
.ob-back-top {
  position: fixed;
  right: 80px;
  bottom: 100px;
  z-index: 6;
  display: flex;
  width: 50px;
  height: 50px;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
  cursor: pointer;
  background-color: #f2f6fc;
  box-shadow: 0 0 6px rgba(0, 0, 0, 0.2);
  &:hover {
    background-color: #fff;
  }
  &__hide {
    display: none;
  }
  .icon {
    width: 26px;
  }
}
</style>