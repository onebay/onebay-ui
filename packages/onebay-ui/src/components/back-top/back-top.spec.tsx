import { mount } from '@vue/test-utils'
import BackTop from './index'

describe('BackTop Snap', () => {
  it('render initial BackTop', () => {
    const wrapper = mount(<BackTop />)
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('BackTop Event', () => {
  it('render initial BackTop', () => {
    const wrapper = mount(<BackTop />)
    wrapper.trigger('click')
    expect(wrapper.emitted()).toHaveProperty('click')
  })
})
