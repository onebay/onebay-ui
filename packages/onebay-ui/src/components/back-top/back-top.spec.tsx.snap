// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackTop Snap render initial BackTop 1`] = `
<div
  class="ob-back-top ob-back-top__hide"
  distance="100"
>
  top
</div>
`;
